api_functional_v4_test.go

```diff
@@ -354,6 +354,66 @@ func TestPutObjectWithMetadata(t *testing.T) {
 	}
 }
 
+// Test put object with streaming signature.
+func TestPutObjectStreaming(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping function tests for short runs")
+	}
+
+	// Seed random based on current time.
+	rand.Seed(time.Now().Unix())
+
+	// Instantiate new minio client object.
+	c, err := NewWithRegion(
+		os.Getenv("S3_ADDRESS"),
+		os.Getenv("ACCESS_KEY"),
+		os.Getenv("SECRET_KEY"),
+		mustParseBool(os.Getenv("S3_SECURE")),
+		"us-east-1",
+	)
+	if err != nil {
+		t.Fatal("Error:", err)
+	}
+
+	// Set user agent.
+	c.SetAppInfo("Minio-go-FunctionalTest", "0.1.0")
+
+	// Generate a new random bucket name.
+	bucketName := randString(60, rand.NewSource(time.Now().UnixNano()),
+		"minio-go-test")
+
+	// Make a new bucket.
+	err = c.MakeBucket(bucketName, "us-east-1")
+	if err != nil {
+		t.Fatal("Error:", err, bucketName)
+	}
+
+	// Upload an object.
+	objectName := "test-object"
+	size := int64(64 * 1024)
```

Comment:
@vadmeste thanks for suggesting the zero byte test case, I caught a bug in the streaming signature implementation. Now that is fixed too.